cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -154,21 +154,39 @@ func NewClusterStateRegistry(cloudProvider cloudprovider.CloudProvider, config C
 	}
 }
 
-// RegisterScaleUp registers scale up.
-func (csr *ClusterStateRegistry) RegisterScaleUp(request *ScaleUpRequest) {
+// RegisterOrUpdateScaleUp registers scaleup for give node group or changes requested node increase
+// count.
+// If increaseDelta is positive then number of new nodes requested is increased; Time and expectedAddTime
+// are reset.
+// If increaseDelete is negative the number of new nodes requested is decreased; Time and expectedAddTime are
+// left intact.
+func (csr *ClusterStateRegistry) RegisterOrUpdateScaleUp(nodeGroupName string, increaseDelta int) {
 	csr.Lock()
 	defer csr.Unlock()
 
-	oldScaleUpRequest, found := csr.scaleUpRequests[request.NodeGroupName]
+	scaleUpRequest, found := csr.scaleUpRequests[nodeGroupName]
 	if !found {
-		csr.scaleUpRequests[request.NodeGroupName] = request
-		return
+		scaleUpRequest = &ScaleUpRequest{
+			NodeGroupName:   nodeGroupName,
+			Time:            time.Now(),
+			ExpectedAddTime: time.Now().Add(csr.config.MaxNodeProvisionTime),
```

Comment:
I am fine with passing `ExpectedAddDuration` as parameter here. Though I believe we should not expose internals of ScaleUpRequest (i.e. the `Time` field to caller of this method). It is implementation detail.

Reply:
Perhaps the problem is that the same exported struct, ScaleUpRequest, is used as part of this methods interface and to keep track of the state internally? I checked and it seems we only use it outside CSR when registering scale-up, and if this is removed (either in favor of the list of parameters, or another struct), it can be made private.